applications/HDF5Application/custom_utilities/container_io_utils.h

```diff
@@ -0,0 +1,477 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:        BSD License
+//                  license: HDF5Application/license.txt
+//
+//  Main author:    Suneth Warnakulasuriya
+//
+
+#pragma once
+
+// System includes
+#include <vector>
+#include <variant>
+#include <type_traits>
+
+// Project includes
+#include "containers/flags.h"
+#include "includes/model_part.h"
+#include "utilities/data_type_traits.h"
+
+// Application includes
+#include "custom_utilities/data_type_utilities.h"
+#include "custom_utilities/vertex.h"
+
+namespace Kratos
+{
+namespace HDF5
+{
+namespace Internals
+{
+
+class FlagIO
+{
+public:
+    ///@name Type definitions
+    ///@{
+
+    template<class TDataType>
+    using TLSType = char;
+
+    template<class TDataType>
+    using ComponentDataType = char;
+
+    ///@}
+    ///@name Public operations
+    ///@{
+
+    template<class TEntityType>
+    inline char GetValue(
+        const TEntityType& rEntity,
+        const Flags& rFlag,
+        char& rTLS) const
```

Comment:
I'm not sure what `TLSType` is for here, but I guess this is one of them

```suggestion
        TLSType<TEntityType>& rTLS) const
```

Reply:
Since it is always char, I don't see a difference in using `char` over `TLSType<TEntityType>`, apart from the fact use of templates makes my head go crazy someties ;)